src/uct/api/uct.h

```diff
@@ -1336,32 +1339,38 @@ ucs_status_t uct_ep_connect_to_ep(uct_ep_h ep, const uct_device_addr_t *dev_addr
  *
  * This routine will create an endpoint for a connection to the remote peer,
  * specified by its socket address.
- * The user may provide private data to be sent on a connection request to the
- * remote peer.
+ * The user may provide a callback function which will be used to fill the
+ * private data that will be sent on a connection request to the remote peer.
+ *
+ * @note In some cases the callback will not be called, for example, if the
+ * endpoint goes into error state before issuing the connection request.
```

Comment:
Is there any more detailed information for this note about when the callback is or is not called? The one example is good, but it seems like there should be more description to know when the callback is guaranteed to fire.

Reply:
we can say it is never guaranteed the callaback will ever fire, @alinask ?

Reply:
@yosefe in which other cases, other than error, it won't be invoked? i mean, if we say that it's never guaranteed, isn't it like saying that a connection may never be established between the client and server?

Reply:
today this is true, but providing a weaker guarantee is forward compatible

Reply:
@gmegan would this work?
"It is never guaranteed that the callaback will be called. If, for example, the endpoint goes into error state before issuing the connection request, the callback will not be invoked."

Also, other than this comment, is everything else ok?

Reply:
That looks good to me. I added some more comments on grammar and usage. Nothing major. Everything else +1.

Reply:
@gmegan I will update them. Thanks!